### PR TITLE
chore(build): uplift several outdated frontend packages

### PR DESCRIPTION
**Pull request by hainenber**
chore(build): uplift several outdated frontend packages

### SUMMARY

Part of doing upkeep for the project on front-end side. Notable changes:
* Upgrade `babel-jest` to current Jest version. This helps reducing 80+ more packages.
* Upgrade `lerna` to latest v8 version. This helps reducing another 80+ more packages.
* Upgrade `react-jsonschema-form` to v5 and switch to `@rjsf` workspace. Also perform appropriate refactoring.
* Remove unused `transform-loader`.
* Upgrade `sinon` to v18 and perform appropriate refactoring.
* Update `prettier` to v3.3.3 in `package.json` and pre-commit's config.

### TESTING INSTRUCTIONS

All CI should pass

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Required feature flags:
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

**Comment by rusackas** [comment]
Hi, I'm re-running CI. Hopefully CI will go green, and we can merge this. Thanks again for your help!

**Comment by hainenber** [comment]
CI is green now 😄

**Comment by rusackas** [comment]
There still seems to be a pre-commit (lining) error to contend with. I'm re-running that CI action in case something went wrong, though :)

**Comment by hainenber** [comment]
> There still seems to be a pre-commit (lining) error to contend with. I'm re-running that CI action in case something went wrong, though :)

Oh that's the one I opened the recent PR to deal with :D

File: superset-frontend/package.json
Comment (rusackas) [inline superset-frontend/package.json:338]: That's quite a bump!!

Comment (hainenber) [reply superset-frontend/package.json:338]: Plenty of accompanying refactors though 😄

**Review by rusackas — APPROVED** [review]
LGTM, pending CI. Really appreciating this sort of work! Hopefully if we keep whittling at all these packages, we'll get to make some of the bigger bumps (i.e. React) before long :)

**Comment by hainenber** [comment]
> LGTM, pending CI. Really appreciating this sort of work! Hopefully if we keep whittling at all these packages, we'll get to make some of the bigger bumps (i.e. React) before long :)

Largest hurdle is Enzyme-to-RTL migration but I've got some of that in store ;)

**Comment by rusackas** [comment]
Looks like we need another `npm install` to get this through... probably my fault for merging dependabot stuff 😓

**Comment by hainenber** [comment]
`npm install`-ed but sadly the CI had a Chrome crash during 1st E2E :(

**Review by justinpark — APPROVED** [review]

**Comment by rusackas** [comment]
It looks like it needs a fresh package-lock file committed. Sorry, these things are hard to keep current! :P

**Comment by hainenber** [comment]
Coming right up with a fresh lock-file :D. Finger-crossed that this one's CI pass 😄 . Hopefully it's gonna be last sync before merge :D

**Comment by hainenber** [comment]
Nice, CI is green now 🥸

**Comment by rusackas** [comment]
Thank you!

**Pull request merged**
